rest-api/api/base_api.py

```diff
@@ -179,6 +179,12 @@ def _get_model_to_update(self, resource, id_, expected_version, participant_id=N
       return self.dao.from_client_json(
           resource, id_=id_, expected_version=expected_version, client_id=app_util.get_oauth_id())
 
+  def _get_patch_model_to_update(self, resource, id_, expected_version, participant_id=None):
+    #pylint: disable=unused-argument
+    # Children of participants accept a participant_id parameter to from_client_json; others don't.
+    if participant_id is not None:
+      return self.dao.get_with_children(id_)
```

Comment:
What's the point of the participant_id is not None check? Should there be an else?

It's worth noting that _get_model_to_update parses JSON resources from the request, rather than reading from the database... I'm not sure why we need another method here, but this is doing something pretty different and might be named differently?